test/fakeintake/client/client.go

```diff
@@ -391,3 +401,19 @@ func (c *Client) GetConnectionsNames() ([]string, error) {
 	}
 	return c.connectionAggregator.GetNames(), nil
 }
+
+// GetProcesses fetches fakeintake on `/api/v1/collector` endpoint and returns
+// all received process payloads
+func (c *Client) GetProcesses() ([]*aggregator.ProcessPayload, error) {
+	err := c.getProcesses()
+	if err != nil {
+		return nil, err
+	}
+
+	var procs []*aggregator.ProcessPayload
+	for _, name := range c.processAggregator.GetNames() {
+		procs = append(procs, c.processAggregator.GetPayloadsByName(name)...)
+	}
+
+	return procs, nil
+}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
As an additional helper, you can add a `FilterProcesses` similar to `FilterLogs` and `FilterMetrics`: it would accept filtering functors and returns only payloads matching filters.
Can be a follow up, only if required by tests

Reply:
Sounds good, we'll likely add it in the future as we flesh out our test suite!